Simplified/AxisService/NYPLAxisPackageService.swift

```diff
@@ -46,79 +48,85 @@ struct NYPLAxisPackageService: NYPLAxisPackageHandling {
   
   /// Downloads the package file and other required files mentioned in the package file (e.g. `Fonts`,
   /// `Images`, `xHTMLs` etc.) requiredfor the book.
-  func downloadPackageContent() {
-    axisItemDownloader.dispatchGroup.wait()
-    self.downloadPackageFile()
-    self.downloadContentFromPackageFile()
+  func downloadPackageContentTasks() -> [NYPLAxisTask] {
+    let packageFileTask = downloadPackageFileTask()
+    let packageContentTask = downloadContentFromPackageFileTask()
+    return [packageFileTask, packageContentTask]
   }
   
   /// Downloads the package file.
-  private func downloadPackageFile() {
-    axisItemDownloader.dispatchGroup.wait()
-    
-    // No need to log error here since axisItemDownloader already logs one.
-    guard axisItemDownloader.shouldContinue else {
-      return
-    }
-    
-    axisItemDownloader.dispatchGroup.enter()
-    
-    guard let endpoint = packageEndpointProvider.getPackageEndpoint()
-    else {
-      axisItemDownloader.leaveGroupAndStopDownload()
-      return
+  private func downloadPackageFileTask() -> NYPLAxisTask {
+    return NYPLAxisTask() { task in
+      
+      guard let endpoint = packageEndpointProvider.getPackageEndpoint()
```

Comment:
Shouldn't properties like `packageEndpointProvider` and `baseURL` etc. need `self.` since this is in a closure? Same for other closures in this file.

Reply:
NYPLAxisPackageService is a struct, and we don't need explicit `self` references in closures for value types when the escaping closure isn't capturing mutating `self`.